### PR TITLE
Http client 0.15

### PR DESCRIPTION
**Pull request by brooksmtownsend**
- Updated to use http-client actor interface (will fail CI/CD since the interface isn't published yet)
- Updated tests accordingly
- Updated dependencies, namely Tokio `0.2` -> `1.1.1`

### CODE REVIEW

**Comment by brooksmtownsend** [comment]
Not ready to be merged until https://github.com/wasmCloud/actor-interfaces/pull/37 is merged

File: http-client/Cargo.toml [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
-name = "wascc-httpclient"
-version = "0.1.0"
-authors = ["Dan Norris <protochron@users.noreply.github.com>"]
+name = "wasmcloud-httpclient"
+version = "0.2.0"
+authors = ["wasmCloud Team"]
 edition = "2018"
-repository = "https://github.com/wascc/http-client-provider"
-description = "HTTP client capability provider for the waSCC wasm host runtime"
+repository = "https://github.com/wasmcloud/http-client-provider"
```
Comment (autodidaddict) [inline http-client/Cargo.toml:6]: this URL is incorrect, it should be https://github.com/wasmcloud/capability-providers

File: http-client/src/lib.rs [hunk 2]
```diff
@@ -9,30 +8,30 @@ extern crate wascc_codec as codec;
...
 use std::collections::HashMap;
 use std::error::Error;
 use std::sync::{Arc, RwLock};
 use std::time::Duration;
 
-const CAPABILITY_ID: &str = "wascc:http_client";
+const CAPABILITY_ID: &str = "wasmcloud:httpclient";
 const VERSION: &str = env!("CARGO_PKG_VERSION");
-const REVISION: u32 = 0;
+const REVISION: u32 = 1;
```
Comment (autodidaddict) [inline http-client/src/lib.rs:24]: as per other reviews, version and revision don't need to be constants anymore.

**Review by autodidaddict — CHANGES_REQUESTED** [review]

File: http-client/src/http_client.rs [hunk 3]
```diff
@@ -52,29 +49,28 @@ fn format_headers(h: &HeaderMap) -> HashMap<String, String> {
...
         }
-        "HEAD" => Ok(client.head(path)),
-        "PUT" => Ok(client.put(path)),
-        "DELETE" => Ok(client.delete(path)),
-        "PATCH" => Ok(client.patch(path)),
-        "OPTIONS" => Ok(client.request(reqwest::Method::OPTIONS, path)),
-        "CONNECT" => Ok(client.request(reqwest::Method::CONNECT, path)),
-        "TRACE" => Ok(client.request(reqwest::Method::TRACE, path)),
+        "HEAD" => Ok(client.head(&req.url)),
+        "PUT" => Ok(client.put(&req.url)),
```
Comment (autodidaddict) [inline http-client/src/http_client.rs:64]: The put doesn't appear to submit a body. `PUT` usually requires content. I know this appears to have been broken in the original version of this provider, too.

File: http-client/src/http_client.rs [hunk 4]
```diff
@@ -52,29 +49,28 @@ fn format_headers(h: &HeaderMap) -> HashMap<String, String> {
...
+) -> Result<RequestBuilder, Box<dyn std::error::Error + Send + Sync>> {
     let mut r = match req.method.as_str() {
         "GET" => {
-            let r = client.get(format!("{}/{}", path, req.query_string).as_str());
+            let r = client.get(&req.url);
             Ok(r)
         }
         "POST" => {
-            let r = client.post(path).body(req.body);
+            let r = client.post(&req.url).body(req.body);
```
Comment (autodidaddict) [inline http-client/src/http_client.rs:60]: This let should've been marked by clippy as superfluous. Same with the one on line 56.

File: http-client/src/lib.rs [hunk 5]
```diff
@@ -143,33 +138,40 @@ impl CapabilityProvider for HttpClientProvider {
...
             OP_BIND_ACTOR if actor == SYSTEM_ACTOR => self.configure(deserialize(msg)?),
             OP_REMOVE_ACTOR if actor == SYSTEM_ACTOR => self.deconfigure(deserialize(msg)?),
             OP_PERFORM_REQUEST => self.request(actor, deserialize(msg)?),
-            OP_GET_CAPABILITY_DESCRIPTOR => self.get_descriptor(),
             _ => Err(format!("Unknown operation: {}", op).into()),
         }
     }
+
+    fn stop(&self) {
+        todo!()
```
Comment (autodidaddict) [inline http-client/src/lib.rs:156]: This will cause a runtime exception when it hits. We should remove and replace with a no-op.

**Review by autodidaddict — CHANGES_REQUESTED** [review]

**Pull request merged**